File: backend/src/main/java/com/snackoverflow/toolgether/domain/user/repository/UserRepository.java
package com.snackoverflow.toolgether.domain.user.repository;

import com.snackoverflow.toolgether.domain.user.entity.User;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {

    Optional<User> findByUsername(String username);
    Optional<User> findByEmail(String email);

    boolean existsByUsername(String username);

    boolean existsByEmail(String email);

    boolean existsByNickname(String nickname);

    Optional<User> findById(long id);

    User findByNickname(String nickname);

    User findByPhoneNumber(String phoneNumber);
}

File: backend/src/main/java/com/snackoverflow/toolgether/domain/user/repository/UserRepository.kt
package com.snackoverflow.toolgether.domain.user.repository;

import com.snackoverflow.toolgether.domain.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;


interface UserRepository : JpaRepository<User, Long> {

    fun findByEmail(email: String): User?

    fun existsByEmail(email: String): Boolean

    @Query("select u from User u where u.id = :id")
    fun findByUserId(id: Long): User?

    fun existsByNickname(nickname: String): Boolean

    fun findByNickname(nickname: String): User?

    @Query("SELECT u FROM User u LEFT JOIN Review r ON r.reviewee = u AND r.createdAt >= :date WHERE r.reviewee IS NULL")
    fun findUsersWithoutReviewsSince(date: LocalDateTime): List<User>

    @Query("SELECT u.profileImage FROM User u WHERE u.profileImage IS NOT NULL")
    fun findAllProfileImageUrl(): List<String>

    fun findByphoneNumber(phoneNumber: String): User?
}